### DISCUSSION TITLE
Open UPPAAL 5.0 dmg on Mac is damaged

### DISCUSSION THREAD

**Discussion by Fnux8890** [UPPAALModelChecker/UPPAAL-Meta#249] [Q&A] [answered]
I'm having trouble opening the dmg and app on Mac.
I'm using the 5.0 version of UPAAL.

Ive tried the instructions about doing the control + click and open from the menu but I still get that its damaged and cant be opened.
Not sure how to fix this, since I've tried to download it again to reinstall it.

Hardware:
      Model Name: MacBook Pro
      Model Identifier: MacBookPro16,1
      Processor Name: 6-Core Intel Core i7
      Processor Speed: 2,6 GHz
      L2 Cache (per Core): 256 KB
      L3 Cache: 12 MB
      Hyper-Threading Technology: Enabled
      Memory: 16 GB
      System Firmware Version: 2020.0.1.0.0 (iBridge: 21.16.365.0.0,0)

MacOS
      ProductName:		macOS
      ProductVersion:		14.0
      BuildVersion:		23A344

<img width="372" alt="image" src="https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/21344589/12d73fd3-38c3-4b49-a216-7ea696572eab">

**Comment by mikucionisaau** [comment]
It's very strange that you cannot start Uppaal. 

Here are a few ideas:
1. Did you drag the `UPPAAL-5.0.0` from `dmg` image into `Applications` or other directory? I don't think it is possible to tun UPPAAL directly from the image, so I am just checking if that is not the case.
2. Have you tried `Control`+click and then `Open` at least twice? It usually shows the `Open` dialog on a second try.
3. In a terminal, visit the containing directory (I'll assume `/Applications`) and try opening from there, like this:
```shell
cd /Applications
open -n UPPAAL-5.0.0.app
```
4. In a terminal, visit the containing directory (I'll assume `/Applications`), remove the quarantine attribute and try opening from there, like this:
```shell
cd /Applications
xattr -r -d com.apple.quarantine UPPAAL-5.0.0.app
open -n UPPAAL-5.0.0.app
```
5. I see that you have macOS version 14.0, the current is 14.3.1 (that's what I have), maybe upgrading or installing updates could help?
6. Check the control sum of `dmg` file by running `md5sum UPPAAL-5.0.0.dmg`, e.g.:
```
$ md5sum UPPAAL-5.0.0.dmg
b61a52e6f38d1446a6d87cf56055c05a  UPPAAL-5.0.0.dmg
```
You should see the same control sum as here.

Please reply how it went.

**Comment by Fnux8890** [comment]
Yep. dragged the dmg into Applications 👍

Tried the:
```bash
open -n UPPAAL-5.0.0.app
```

and

```bash
cd /Applications
xattr -r -d com.apple.quarantine UPPAAL-5.0.0.app
open -n UPPAAL-5.0.0.app
```

produced the same result.

I get
```bash
md5sum: UPPAAL-5.0.0.dmg: No such file or directory
```
from the md5sum.

will try to update my system to see if that helps 👍

**Comment by Fnux8890** [comment]
Okay. After updating I got the same response.
I still cant seem to open it

**Comment by mikucionisaau** [comment, marked as answer]
The last command should be done in your Downloads, like:
```shell
cd Downloads
md5sum UPPAAL-5.0.0.dmg
```

Do you see more attributes on UPPAAL app? What is the output from the following command?
```shell
xattr -l /Applications/UPPAAL-5.0.0.app
```


One more idea: [Privacy and Security](https://support.apple.com/en-mide/102445) in Settings
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/4868894/4d263857-5c09-4a06-bb3c-fc7dad3b0cb9)
What option do you have selected there?
Can you find UPPAAL app and click `Open Anyway`?

The issue is that Apple does this to applications which are not notarized.

**Comment by Fnux8890** [comment]
My md5sum result in Downloads
```sh
md5sum UPPAAL-5.0.0.dmg
b61a52e6f38d1446a6d87cf56055c05a  UPPAAL-5.0.0.dmg
```
My xattr results
```sh
xattr -l UPPAAL-5.0.0.app
com.apple.macl:
com.apple.provenance:
```
Dont know id this information could be important for later user.

but I got the open Anyways now after updating 🙌

Thanks so much mate
